extensions/info/deployment/src/main/java/io/quarkus/info/deployment/InfoProcessor.java

```diff
@@ -0,0 +1,142 @@
+
+    @BuildStep(onlyIf = GitInInfoEndpointEnabled.class)
+    InfoBuildTimeValuesBuildItem gitInfo(CurateOutcomeBuildItem curateOutcomeBuildItem,
+            OutputTargetBuildItem outputTargetBuildItem) {
+        File projectRoot = highestKnownProjectDirectory(curateOutcomeBuildItem, outputTargetBuildItem);
+        if (projectRoot == null) {
+            log.debug("Unable to determine project directory");
+            return null;
+        }
+        RepositoryBuilder repositoryBuilder = new RepositoryBuilder().findGitDir(projectRoot);
+        if (repositoryBuilder.getGitDir() == null) {
+            log.debug("Project is not checked in to git");
+            return null;
+        }
+        try (Repository repository = repositoryBuilder.build()) {
+
+            RevCommit latestCommit = new Git(repository).log().setMaxCount(1).call().iterator().next();
+
+            PersonIdent authorIdent = latestCommit.getAuthorIdent();
+            Date authorDate = authorIdent.getWhen();
+            TimeZone authorTimeZone = authorIdent.getTimeZone();
+
+            Map<String, Object> data = new LinkedHashMap<>();
+            data.put("branch", repository.getBranch());
+            data.put("commit", Map.of(
+                    "id", latestCommit.getName(),
+                    "time",
+                    ISO_OFFSET_DATE_TIME.format(
+                            OffsetDateTime.ofInstant(authorDate.toInstant(), authorTimeZone.toZoneId()))));
+            return new InfoBuildTimeValuesBuildItem("git", data);
+        } catch (Exception e) {
+            log.debug("Unable to determine git information", e);
+            return null;
+        }
+    }
+
+    private File highestKnownProjectDirectory(CurateOutcomeBuildItem curateOutcomeBuildItem,
+            OutputTargetBuildItem outputTargetBuildItem) {
+        ApplicationModel applicationModel = curateOutcomeBuildItem.getApplicationModel();
+        WorkspaceModule workspaceModule = applicationModel.getAppArtifact().getWorkspaceModule();
+        if (workspaceModule != null) {
+            // in this case we know the precise project root
+            return workspaceModule.getModuleDir();
+        }
+        // in this case we will simply use the build directory and let jgit go up the file system to determine the git directory - if any
+        return outputTargetBuildItem.getOutputDirectory().toFile();
+    }
+
+    @BuildStep(onlyIf = BuildInInfoEndpointEnabled.class)
+    InfoBuildTimeValuesBuildItem buildInfo(CurateOutcomeBuildItem curateOutcomeBuildItem, InfoBuildTimeConfig config) {
+        ApplicationModel applicationModel = curateOutcomeBuildItem.getApplicationModel();
+        ResolvedDependency appArtifact = applicationModel.getAppArtifact();
+        Map<String, Object> buildData = new LinkedHashMap<>();
+        buildData.put("group", appArtifact.getGroupId());
+        buildData.put("artifact", appArtifact.getArtifactId());
```

Comment:
You're following some Spring conventions that you are not using `groupId` and `artifactId`? I think that's generally how we call them.

Reply:
It's a bit strange that it follows neither Maven nor Gradle (group, name, version) terminology.

Reply:
Maybe that's the exact reason it was chosen?